sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -601,6 +601,264 @@ manager, with `__aenter__`, `__aexit__`, and `close` methods.
+        if status.lower() == "validationfailed":
+            raise HttpResponseError(response=response, error=body["error"])
+        return status
+```
+
+To plug into a client method which was generated as an LRO, you can pass the `polling` keyword argument. The `polling`
+keyword argument takes a `azure.core.polling.PollingMethod` and allows for a custom strategy to be passed in to the keyword
+argument `lro_algorithms`:
+
+```python
+class ServiceOperations:
+
+    def begin_analyze(self, data: AnyStr, name: str) -> LROPoller[JSON]:
+        return self._client.begin_analyze(
+            data,
+            name,
+            polling=LROBasePolling(
+                lro_algorithms=[
+                    CustomOperationResourcePolling()  # overrides other LRO strategies
+                ]
+            ),
+            **kwargs
+        )
+```
+
+This example uses the default polling method - `LROBasePolling` and just overrides the algorithm used for polling.
+If you need to control the polling loop, then see the next section.
+
+
+### Polling method - LROBasePolling/AsyncLROBasePolling
+
+Built-in methods for polling are included in `azure-core` as both sync / async variants - `LROBasePolling` and `AsyncLROBasePolling`.
+You can also use `azure.core.polling.NoPolling` which will not initiate polling and simply return the 
+deserialized initial response when called with `poller.result()`.
+
+To use `NoPolling`, you can simply pass `polling=False` to an operation generated as an LRO:
+
+```python
+class ServiceOperations:
+
+    def begin_analyze(self, data: AnyStr, name: str) -> LROPoller[JSON]:
+        return self._client.begin_analyze(
+            data,
+            name,
+            polling=False,
+            **kwargs
+        )
+```
+
+The polling method runs the polling loop and performs GET requests to the status monitor to check if a terminal state is reached. 
+In between polls it inserts delay based on the service sent `retry-after` header or the given `polling_interval`.
+
+To customize parts of the polling method, you can create a subclass which uses `LROBasePolling` and overrides necessary methods.
+If significant customization is necessary, use `azure.core.polling.PollingMethod` and implement all the necessary methods.
+
+Example: Create an LRO method which will poll for when a file gets uploaded successfully (greatly simplified)
+
+
+```python
+class CustomLROPolling(PollingMethod):
+    def __init__(self, polling_interval=30, **kwargs) -> None:
+        self._polling_interval = polling_interval
+        self._resource = None
+        self._finished = False
+        self._status = "InProgress"
+        self._kwargs = kwargs
+
+    def initialize(self, client, initial_response, deserialization_callback):
+        """Set the initial status of this LRO.
+        """ 
+        self.client = client
+        self.file_id = response.file_id
+        self._pipeline_response = self._initial_response = initial_response
```

Comment:
We never use `self._pipeline_response`?
I also didn't think it was the pipeline response that was passed into the polling method? But I probably misremember...

Reply:
For standard LROs, it is a PipelineResponse that is passed in to initialize(...). We later use its `context` to restore the transport:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/polling/base_polling.py#L461

But you're right that it's not used in this code and we actually pass the deserialized initial_response in the example... which is likely not pickle-able, nor does it have `context`. So I'll need to fix this.